### DISCUSSION TITLE
npm run build errors

### DISCUSSION THREAD

**Discussion by MJB27** [google/model-viewer#3088] [Q&A]
I am currently running model-viewer 1.9.1 on MacOS Monterey

I seem to be getting errors when running the cmd 'npm run build' after running 'npm run bootstrap' which has no errors within vs studio code editor using the  repository. 

I have followed the instruction from the read me file and used npm version 6.14. please see errors below as I am stuck and unable to find out what's going wrong. could some help or assist me in what I am doing wrong or what I need to do. 

> model-viewer-packages@ build /Users/mjb/Desktop/Documents /model-viewer-1.9.1
> lerna run build --stream

lerna notice cli v3.22.1
lerna info versioning independent
lerna info Executing command in 4 packages: "npm run build"
@google/model-viewer: > @google/model-viewer@1.9.0 build /Users/mjb/Desktop/Documents /model-viewer-1.9.1/packages/model-viewer
@google/model-viewer: > npm run build:tsc && npm run build:rollup
@google/model-viewer: > @google/model-viewer@1.9.0 build:tsc /Users/mjb/Desktop/Documents /model-viewer-1.9.1/packages/model-viewer
@google/model-viewer: > tsc --incremental
@google/model-viewer: error TS2688: Cannot find type definition file for 'accepts'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'accepts'
@google/model-viewer: error TS2688: Cannot find type definition file for 'babel__core'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'babel__core'
@google/model-viewer: error TS2688: Cannot find type definition file for 'babel__generator'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'babel__generator'
@google/model-viewer: error TS2688: Cannot find type definition file for 'babel__template'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'babel__template'
@google/model-viewer: error TS2688: Cannot find type definition file for 'babel__traverse'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'babel__traverse'
@google/model-viewer: error TS2688: Cannot find type definition file for 'body-parser'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'body-parser'
@google/model-viewer: error TS2688: Cannot find type definition file for 'browserslist-useragent'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'browserslist-useragent'
@google/model-viewer: error TS2688: Cannot find type definition file for 'caniuse-api'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'caniuse-api'
@google/model-viewer: error TS2688: Cannot find type definition file for 'command-line-args'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'command-line-args'
@google/model-viewer: error TS2688: Cannot find type definition file for 'command-line-usage'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'command-line-usage'
@google/model-viewer: error TS2688: Cannot find type definition file for 'component-emitter'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'component-emitter'
@google/model-viewer: error TS2688: Cannot find type definition file for 'connect'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'connect'
@google/model-viewer: error TS2688: Cannot find type definition file for 'cors'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'cors'
@google/model-viewer: error TS2688: Cannot find type definition file for 'http-assert'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'http-assert'
@google/model-viewer: error TS2688: Cannot find type definition file for 'keygrip'.
@google/model-viewer:   The file is in the program because:
@google/model-viewer:     Entry point for implicit type library 'keygrip'
@google/model-viewer: npm ERR! Darwin 21.2.0
@google/model-viewer: npm ERR! argv "/Users/mjb/Desktop/Documents /model-viewer-1.9.1/node_modules/node/bin/node" "/usr/local/bin/npm" "run" "build:tsc"
@google/model-viewer: npm ERR! node v16.13.0
@google/model-viewer: npm ERR! npm  v3.10.10
@google/model-viewer: npm ERR! code ELIFECYCLE
@google/model-viewer: npm ERR! @google/model-viewer@1.9.0 build:tsc: `tsc --incremental`
@google/model-viewer: npm ERR! Exit status 2
@google/model-viewer: npm ERR! 
@google/model-viewer: npm ERR! Failed at the @google/model-viewer@1.9.0 build:tsc script 'tsc --incremental'.
@google/model-viewer: npm ERR! Make sure you have the latest version of node.js and npm installed.
@google/model-viewer: npm ERR! If you do, this is most likely a problem with the @google/model-viewer package,
@google/model-viewer: npm ERR! not with npm itself.
@google/model-viewer: npm ERR! Tell the author that this fails on your system:
@google/model-viewer: npm ERR!     tsc --incremental
@google/model-viewer: npm ERR! You can get information on how to open an issue for this project with:
@google/model-viewer: npm ERR!     npm bugs @google/model-viewer
@google/model-viewer: npm ERR! Or if that isn't available, you can get their info via:
@google/model-viewer: npm ERR!     npm owner ls @google/model-viewer
@google/model-viewer: npm ERR! There is likely additional logging output above.
@google/model-viewer: npm ERR! Please include the following file with any support request:
@google/model-viewer: npm ERR!     /Users/mjb/Desktop/Documents /model-viewer-1.9.1/packages/model-viewer/npm-debug.log
@google/model-viewer: npm ERR! Darwin 21.2.0
@google/model-viewer: npm ERR! argv "/Users/mjb/Desktop/Documents /model-viewer-1.9.1/node_modules/node/bin/node" "/usr/local/bin/npm" "run" "build"
@google/model-viewer: npm ERR! node v16.13.0
@google/model-viewer: npm ERR! npm  v3.10.10
@google/model-viewer: npm ERR! code ELIFECYCLE
@google/model-viewer: npm ERR! @google/model-viewer@1.9.0 build: `npm run build:tsc && npm run build:rollup`
@google/model-viewer: npm ERR! Exit status 1
@google/model-viewer: npm ERR! 
@google/model-viewer: npm ERR! Failed at the @google/model-viewer@1.9.0 build script 'npm run build:tsc && npm run build:rollup'.
@google/model-viewer: npm ERR! Make sure you have the latest version of node.js and npm installed.
@google/model-viewer: npm ERR! If you do, this is most likely a problem with the @google/model-viewer package,
@google/model-viewer: npm ERR! not with npm itself.
@google/model-viewer: npm ERR! Tell the author that this fails on your system:
@google/model-viewer: npm ERR!     npm run build:tsc && npm run build:rollup
@google/model-viewer: npm ERR! You can get information on how to open an issue for this project with:
@google/model-viewer: npm ERR!     npm bugs @google/model-viewer
@google/model-viewer: npm ERR! Or if that isn't available, you can get their info via:
@google/model-viewer: npm ERR!     npm owner ls @google/model-viewer
@google/model-viewer: npm ERR! There is likely additional logging output above.
@google/model-viewer: npm ERR! Please include the following file with any support request:
@google/model-viewer: npm ERR!     /Users/mjb/Desktop/Documents /model-viewer-1.9.1/packages/model-viewer/npm-debug.log
lerna ERR! npm run build exited 1 in '@google/model-viewer'

npm ERR! Darwin 21.2.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
npm ERR! node v6.14.0
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! model-viewer-packages@ build: `lerna run build --stream`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the model-viewer-packages@ build script 'lerna run build --stream'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the model-viewer-packages package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     lerna run build --stream
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs model-viewer-packages
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls model-viewer-packages
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/mjb/Desktop/Documents /model-viewer-1.9.1/npm-debug.log
sh-3.2$

**Comment by ghost** [comment]
Thanks for the feedback, upgrading to Monterey then I will try to reproduce this.
In the mean time try opening the model-viewer folder and run `npm run build` from within the folder.
for me that is:
```
cd ~/repos/model-viewer
npm run build
```

**Comment by MJB27** [comment]
Thanks for getting back to me on this issue. I have tried what you have recommend by running npm run build. I am still getting the errors. Please see below:

npm ERR! Darwin 21.2.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
npm ERR! node v6.14.0
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! @google/model-viewer@1.9.0 build: `npm run build:tsc && npm run build:rollup`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google/model-viewer@1.9.0 build script 'npm run build:tsc && npm run build:rollup'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @google/model-viewer package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build:tsc && npm run build:rollup
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @google/model-viewer
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @google/model-viewer
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/mjb/Desktop/Documents /model-viewer-1.9.1/packages/model-viewer/npm-debug.log

**Reply by elalish** [reply]
@MJB27 I believe node v6 and npm v3 are  both very old. Have you tried updating? I believe  we currently build with node v16.5 and npm v7.19.

**Reply by Timsline** [reply]
Hello @elalish, 
I have node v16.5 and npm v7.19, but I am still having an error on npm run build :
![image](https://user-images.githubusercontent.com/45558719/147821819-789d6261-b5d3-4b30-ac4e-94a574b7948f.png)

> build
> ./node_modules/.bin/lerna run build --stream

lerna notice cli v4.0.0
lerna info versioning independent
lerna info Executing command in 4 packages: "npm run build"
@google/model-viewer: > @google/model-viewer@1.9.0 build
@google/model-viewer: > npm run build:tsc && npm run build:rollup
@google/model-viewer: > @google/model-viewer@1.9.0 build:tsc
@google/model-viewer: > tsc --incremental
@google/model-viewer: node_modules/@types/mocha/index.d.ts(2584,13): error TS2300: Duplicate identifier 'beforeEach'.
@google/model-viewer: node_modules/@types/mocha/index.d.ts(2602,13): error TS2300: Duplicate identifier 'afterEach'.
@google/model-viewer: node_modules/@types/mocha/index.d.ts(2618,13): error TS2300: Duplicate identifier 'describe'.
@google/model-viewer: node_modules/@types/mocha/index.d.ts(2639,13): error TS2300: Duplicate identifier 'xdescribe'.
@google/model-viewer: node_modules/@types/mocha/index.d.ts(2653,13): error TS2300: Duplicate identifier 'it'.
@google/model-viewer: node_modules/@types/mocha/index.d.ts(2674,13): error TS2300: Duplicate identifier 'xit'.
@google/model-viewer: ../../../../@types/jasmine/index.d.ts(33,18): error TS2300: Duplicate identifier 'describe'.
@google/model-viewer: ../../../../@types/jasmine/index.d.ts(47,18): error TS2300: Duplicate identifier 'xdescribe'.
@google/model-viewer: ../../../../@types/jasmine/index.d.ts(56,18): error TS2300: Duplicate identifier 'it'.
@google/model-viewer: ../../../../@types/jasmine/index.d.ts(72,18): error TS2300: Duplicate identifier 'xit'.
@google/model-viewer: ../../../../@types/jasmine/index.d.ts(100,18): error TS2300: Duplicate identifier 'beforeEach'.
@google/model-viewer: ../../../../@types/jasmine/index.d.ts(107,18): error TS2300: Duplicate identifier 'afterEach'.
lerna ERR! npm run build exited 2 in '@google/model-viewer'

Could you help please ? 
(PS : I am using an Ubuntu distribution generated by wsl.exe)

**Reply by elalish** [reply]
It looks like you might just be in a bad state. When  that happens I  usually run `npm run clean`, delete all the `node_modules` directories, then run `npm run  bootstrap` and then `npm run  build`. If that doesn't help, maybe just re-clone.

**Reply by Timsline** [reply]
Thanks, but now I have an error, model-viewer prevent bootstrap to execute :
![image](https://user-images.githubusercontent.com/45558719/148283778-ddd77a3a-4496-4ae6-af08-1d12c1bf6792.png)
Does model-viewer work on Linux VM or should it be installed in dualboot ?

**Reply by elalish** [reply]
I recently fixed that shared-assets thing; are you sure you've pulled/cloned recently?

**Reply by Timsline** [reply]
Yes, thank you very much, I had the repo before you pushed, model-viewer is working except that 3D models are not animated yet.
But I'm gonna try find my way through it 
Thanks & happy holidays
